packages/headless/src/controllers/commerce/facets/core/generator/headless-commerce-facet-generator.ts

```diff
@@ -60,7 +66,7 @@ export interface CommerceFacetGeneratorOptions {
   buildRegularFacet: CommerceFacetBuilder<CommerceRegularFacet>;
   buildNumericFacet: CommerceFacetBuilder<CommerceNumericFacet>;
   buildDateFacet: CommerceFacetBuilder<CommerceDateFacet>;
-  // TODO: buildCategoryFacet: CommerceFacetBuilder<CommerceCategoryFacet>;
+  buildCategoryFacet: CommerceCategoryFacetBuilder;
```

Comment:
Why not this?
```suggestion
  buildCategoryFacet: CommerceFacetBuilder<CommerceCategoryFacet>;
```

Reply:
Because `CommerceCategoryFacet` doesn't expose all of the methods of the `CoreCommerceFacet`.

I could instead tweak the `CommerceFacetBuilder` type like so, though:

```typescript
type CommerceFacetBuilder<
  Facet extends Omit<
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    CoreCommerceFacet<any, any>,
    | 'isValueExcluded'
    | 'toggleExclude'
    | 'toggleSingleExclude'
    | 'toggleSingleSelect'
  >,
> = (engine: CommerceEngine, options: CommerceFacetOptions) => Facet;
```

It's a bit convoluted, but more consistent.